java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/ai/openai/chatcompletion/OpenAIChatCompletion.java

```diff
@@ -0,0 +1,129 @@
+
+import com.azure.ai.openai.models.ChatCompletionsOptions;
+import com.azure.ai.openai.models.ChatMessage;
+import com.azure.ai.openai.models.ChatRole;
+import com.microsoft.openai.OpenAIAsyncClient;
+import com.microsoft.semantickernel.Verify;
+import com.microsoft.semantickernel.ai.AIException;
+import com.microsoft.semantickernel.chatcompletion.ChatCompletion;
+import com.microsoft.semantickernel.chatcompletion.ChatHistory;
+import com.microsoft.semantickernel.chatcompletion.ChatRequestSettings;
+import com.microsoft.semantickernel.connectors.ai.openai.azuresdk.ClientBase;
+
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+/** OpenAI chat completion client. */
+public class OpenAIChatCompletion extends ClientBase implements ChatCompletion<OpenAIChatHistory> {
+    public OpenAIChatCompletion(OpenAIAsyncClient client, String modelId) {
+        super(client, modelId);
+    }
+
+    public static class Builder implements ChatCompletion.Builder {
+        public Builder() {}
+
+        public ChatCompletion<OpenAIChatHistory> build(OpenAIAsyncClient client, String modelId) {
+            return new OpenAIChatCompletion(client, modelId);
+        }
+    }
+
+    public Mono<String> generateMessageAsync(
+            OpenAIChatHistory chat, @Nullable ChatRequestSettings requestSettings) {
+
+        if (requestSettings == null) {
+            requestSettings = new ChatRequestSettings();
+        }
+        return this.internalGenerateChatMessageAsync(chat, requestSettings);
+    }
+
+    /**
+     * Generate a new chat message
+     *
+     * @param chat Chat history
+     * @param requestSettings AI request settings
+     * @return
+     */
+    private Mono<String> internalGenerateChatMessageAsync(
+            ChatHistory chat, ChatRequestSettings requestSettings) {
+        Verify.notNull(chat);
+        Verify.notNull(requestSettings);
+
+        validateMaxTokens(requestSettings.getMaxTokens());
+        ChatCompletionsOptions options = createChatCompletionsOptions(requestSettings, chat);
+
+        return getClient()
+                .getChatCompletions(getModelId(), options)
+                .flatMap(
+                        response -> {
+                            if (response == null || response.getChoices().isEmpty()) {
+                                return Mono.error(
+                                        new AIException(
+                                                AIException.ErrorCodes.InvalidResponseContent,
+                                                "Chat completions not found"));
+                            } else {
+                                return Mono.just(
+                                        response.getChoices().get(0).getMessage().getContent());
+                            }
+                        });
+    }
+
+    private static ChatCompletionsOptions createChatCompletionsOptions(
```

Comment:
This is an OpenAI specific implementation so we should reflect that in the classname